app/src/components/card/cardImage.tsx

```diff
@@ -0,0 +1,64 @@
+import Image from "next/image";
+import React from "react";
+import { useEffect, useRef, useState } from "react";
+import { isCollectionCardDataType } from "./card";
+
+const CardImage = ({ record }) => {
+  const [imageHeight, setImageHeight] = useState(144);
+  const imageRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    const updateImageHeight = () => {
+      if (imageRef.current) {
+        const image = imageRef.current as HTMLElement;
+        const imageWidth = image.offsetWidth;
+
+        setImageHeight(0.5 * imageWidth);
+      }
+    };
+    updateImageHeight();
+    window.addEventListener("resize", updateImageHeight);
+
+    return () => window.removeEventListener("resize", updateImageHeight);
+  }, []);
+
+  const [imageSrc, setImageSrc] = useState(
+    record.imageID ? record.imageURL : "/noImage.png"
+  );
+  const isCollection = isCollectionCardDataType(record);
+  return (
+    <div
+      ref={imageRef}
+      style={{
+        overflow: "hidden",
+        height: imageHeight,
+      }}
+    >
+      <Image
+        src={imageSrc}
+        alt=""
+        id={
+          record.imageID
+            ? isCollection
+              ? `image-${record.title}-${record.imageID}`
```

Comment:
The `.title` may not be a slug so it will not work for the `id` attribute (`id`s cannot have white space).